### PR TITLE
add task schema example: install-os-common + obm-control

### PR DESCRIPTION
**Pull request by yyscamper**
This is the initial PR for [Task-Schema](https://github.com/RackHD/RackHD/wiki/proposal-Task-Schema).

The primary purpose for this PR is to show some example about how the Task-Schema looks like and how the Task & Schema & Job is linked. I choose the most complex task `install-os` and a simple `obm-control` as example.

If the Task-Schema example is done, then the [Task-Annotation](https://github.com/RackHD/RackHD/wiki/proposal-task-annotation) can carry on in parallel.

Meanwhile, You should know that expect the unit-test, no other code will use the schema, so this PR is safe and will not impact our existing function. We will have subsequent PRs to gradually enrich the Task-Schema functionality.

@RackHD/corecommitters @WangWinson @iceiilin @cgx027 @pengz1 @leoyjchang @zyoung51 @jlongever

### CODE REVIEW

**Comment by WangWinson** [comment]
:+1: 
This is a very good start.

File: lib/task-data/schemas/rackhd-task-schema.json [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+    "description": "The extension for RackHD tasks schema, all RackHD tasks schema should adhere to this schema.",
+    "allOf": [
+        {
+            "$ref": "http://json-schema.org/draft-04/schema"
+        }
+    ],
+    "definitions": {
+        "describeJob": {
+            "type": "string",
+            "description": "This property is required to point to a job's di injectable name."
```
Comment (anhou) [inline lib/task-data/schemas/rackhd-task-schema.json]: don't need 'This property is required to point to' ?

**Comment by anhou** [comment]
:+1:

File: lib/task-data/schemas/rackhd-task-schema.json (hunk 1 above)
Comment (yyscamper) [inline lib/task-data/schemas/rackhd-task-schema.json]: I think either is OK.
I see redfish schema has similar word, you can see the `readonly` description which is copied from redfish schema.

**Comment by yyscamper** [comment]
Since there are some other stories depend on this PR and consider this is a very safe PR as the existing functionality will not be impacted. So I will merge this first after collecting enough approvals. If anyone has any comment, I will open subsequent PR to fix.

**Pull request merged**